**Fin 585R**  
**Diether**  
**Problem Set**  
**Time Series Tests of the CAPM**  

**Overview**  

In this problem set you test the CAPM using the time series framework. Specifically, you test whether the CAPM holds with respect to *momentum portfolios* (Jegadeesh and Titman, 1993, "Returns to Buying Winners and Selling Losers: Implications for Stock Market Efficiency") using the **time series testing framework.** Momentum portfolios are formed based on past returns. Specifically, momentum portfolios are most commonly formed based on the cumulative return from months $t-12$ to $t-2$:

$$
r_{i,t-12:t-2} \approx \sum_{x=2}^{12} \log(1+r_{i,t-x})
$$

I formed equal-weight momentum portfolios for you; the portfolios are available for download on *Learning Suite* or you can download it directly from the following link: [momentum portfolios](https://diether.org/prephd/10-port_mom_ew.csv)). Note, that in my data the returns for these portfolios are in percent; I have multiplied the raw returns by 100. Also, to test the CAPM you are going to need a proxy for the market portfolio and for the riskfree rate. Data from these can be found at [Ken French's Data Library](https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html). For your convenience I have created a csv file that contains both these variables, and it can be loaded directly into a dataframe from my website (see the code below). The `dataframe` contains the excess return on a proxy for the market portfolio (`exmkt`), a proxy for the riskfree rate (`rf`), and some other portfolios you can ignore. The returns from Ken French's library are also in percent: raw returns multiplied by 100.

For questions that require some write-up, create a markdown cell (use the Cell Toolbar)  and write your answer in the markdown cell (this cell is a markdown cell and here is a [markdown cheat sheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)). 

**Tasks and Questions**  

1. Compute and report the sample mean, standard deviation, and t-test for the mean = 0 for the *excess returns* of the momentum portfolios ($r_p-r_f$). <br><br>

2. Test the CAPM by running a time series CAPM regression for each of the momentum portfolios: <br><br>
$$
r_{pt} - r_{ft} = \alpha_p + \beta_{pM}( r_{Mt} - r_{ft}) + \epsilon_{it}
$$
<br>Consolidate all your regression results into one table (hint, go take a look at the jupyter notebook on estimating regressions in `pandas` and `statsmodels`). <br><br>

3. Interpret the regression results from question 2). What can you infer? Can you reject that
the CAPM holds? Is the market portfolio the tangency portfolio? <br><br>

4. Create a spread portfolio:<br><br>
$$
r_{spread,t} = r_{4t} - r_{0t}.
$$
<br>Test the CAPM using this portfolio. Can you reject the CAPM?<br><br>

5. Estimate the security market line using the data available for this homework. Specifically, estimate the following line:  <br><br>
$$
E(r_p) = r_f + \beta_{p}\bigl[E(r_M) - r_f\bigr]
$$
<br> You don't need to plot the estimated line, but report your estimates of $r_f$ and $E(r_M) - r_f$. <br><br>

6. Why is the intercept in time series CAPM regression called an *average abnormal return*? Briefly explain.

In [14]:
import pandas as pd
import numpy as np
from finance_byu.summarize import summary
import statsmodels.formula.api as smf
from finance_byu.regtables import Regtable

In [2]:
port = pd.read_csv('https://diether.org/prephd/10-port_mom_ew.csv', parse_dates=['caldt'])
port

caldt         p0         p1         p2        p3        p4
0    1927-01-31  -2.708397   4.290747   0.776512  1.210773  0.508464
1    1927-02-28   5.418420   8.356758   4.758203  4.905527  5.422874
2    1927-03-31  -3.829973  -2.910900  -1.801945  0.259834  0.020529
3    1927-04-30  -0.747539  -0.438827  -0.216951 -0.172500  3.216826
4    1927-05-31   3.314240   5.759958   7.950440  8.014930  8.587076
...         ...        ...        ...        ...       ...       ...
1144 2022-05-31  -5.392614  -2.553349   0.502323 -0.183222  1.199527
1145 2022-06-30  -5.344633  -8.743887  -8.291228 -4.778448 -9.131728
1146 2022-07-29  11.776372  11.033183   9.284861  6.875124  8.193744
1147 2022-08-31   0.317134  -2.100848  -2.140245 -2.222316  0.123339
1148 2022-09-30 -10.184062 -10.091326 -10.592417 -5.988948 -8.171841

[1149 rows x 6 columns]

In [3]:
fac = pd.read_csv('https://diether.org/prephd/10-factors.csv', parse_dates=['caldt'])
fac

caldt  exmkt   smb   hml   umd    rf
0    1927-01-31  -0.06 -0.37  4.54  0.36  0.25
1    1927-02-28   4.18  0.04  2.94 -2.14  0.26
2    1927-03-31   0.13 -1.65 -2.61  3.61  0.30
3    1927-04-30   0.46  0.30  0.81  4.30  0.25
4    1927-05-31   5.44  1.53  4.73  3.00  0.30
...         ...    ...   ...   ...   ...   ...
1144 2022-05-31  -0.34 -1.85  8.41  2.48  0.03
1145 2022-06-30  -8.43  2.09 -5.97  0.79  0.06
1146 2022-07-29   9.57  2.81 -4.10 -3.96  0.08
1147 2022-08-31  -3.77  1.39  0.31  2.10  0.19
1148 2022-09-30  -9.35 -0.82  0.03  3.48  0.19

[1149 rows x 6 columns]

### 1.
Compute and report the sample mean, standard deviation, and t-test for the mean = 0 for the *excess returns* of the momentum portfolios ($r_p-r_f$).

In [ ]:
# port = port.merge(fac, on='caldt')
# port[['er0', ...]] = port[['p0', ...]].sub(port['rf'], axis='index')

In [15]:
port['exmkt'] = fac['exmkt']
port['er0'] = port['p0'] - fac['rf']
port['er1'] = port['p1'] - fac['rf']
port['er2'] = port['p2'] - fac['rf']
port['er3'] = port['p3'] - fac['rf']
port['er4'] = port['p4'] - fac['rf']
port

caldt         p0         p1         p2        p3        p4  \
0    1927-01-31  -2.708397   4.290747   0.776512  1.210773  0.508464   
1    1927-02-28   5.418420   8.356758   4.758203  4.905527  5.422874   
2    1927-03-31  -3.829973  -2.910900  -1.801945  0.259834  0.020529   
3    1927-04-30  -0.747539  -0.438827  -0.216951 -0.172500  3.216826   
4    1927-05-31   3.314240   5.759958   7.950440  8.014930  8.587076   
...         ...        ...        ...        ...       ...       ...   
1144 2022-05-31  -5.392614  -2.553349   0.502323 -0.183222  1.199527   
1145 2022-06-30  -5.344633  -8.743887  -8.291228 -4.778448 -9.131728   
1146 2022-07-29  11.776372  11.033183   9.284861  6.875124  8.193744   
1147 2022-08-31   0.317134  -2.100848  -2.140245 -2.222316  0.123339   
1148 2022-09-30 -10.184062 -10.091326 -10.592417 -5.988948 -8.171841   

            er0        er1        er2       er3       er4  exmkt  
0     -2.958397   4.040747   0.526512  0.960773  0.258464  -0.06  
1      5.158420   8.096758   4.498203  4.645527  5.162874   4.18  
2     -4.129973  -3.210900  -2.101945 -0.040166 -0.279471   0.13  
3     -0.997539  -0.688827  -0.466951 -0.422500  2.966826   0.46  
4      3.014240   5.459958   7.650440  7.714930  8.287076   5.44  
...         ...        ...        ...       ...       ...    ...  
1144  -5.422614  -2.583349   0.472323 -0.213222  1.169527  -0.34  
1145  -5.404633  -8.803887  -8.351228 -4.838448 -9.191728  -8.43  
1146  11.696372  10.953183   9.204861  6.795124  8.113744   9.57  
1147   0.127134  -2.290848  -2.330245 -2.412316 -0.066661  -3.77  
1148 -10.374062 -10.281326 -10.782417 -6.178948 -8.361841  -9.35  

[1149 rows x 12 columns]

In [16]:
summary(port[['er0', 'er1', 'er2', 'er3', 'er4']]).loc[['mean', 'std', 'tstat']].round(2)

er0      er1      er2      er3      er4
count  1149.00  1149.00  1149.00  1149.00  1149.00
mean      0.13     0.59     0.83     1.01     1.32
std       9.13     7.10     6.28     5.89     6.65
tstat     0.47     2.80     4.49     5.80     6.71
pval      0.64     0.01     0.00     0.00     0.00
min     -39.68   -33.95   -31.78   -30.77   -31.74
25%      -3.85    -2.58    -1.84    -1.80    -2.03
50%       0.12     0.87     1.18     1.47     1.88
75%       3.89     3.69     3.60     4.08     5.23
max      94.40    57.62    50.36    42.07    38.00

### 2.
Test the CAPM by running a time series CAPM regression for each of the momentum portfolios:
<br>
$$
r_{pt} - r_{ft} = \alpha_p + \beta_{pM}( r_{Mt} - r_{ft}) + \epsilon_{it}
$$
<br>
Consolidate all your regression results into one table (hint, go take a look at the jupyter notebook on estimating regressions in `pandas` and `statsmodels`).

In [20]:
reg0 = smf.ols('er0 ~ exmkt', data=port).fit()
reg1 = smf.ols('er1 ~ exmkt', data=port).fit()
reg2 = smf.ols('er2 ~ exmkt', data=port).fit()
reg3 = smf.ols('er3 ~ exmkt', data=port).fit()
reg4 = smf.ols('er4 ~ exmkt', data=port).fit()

In [21]:
tbl = Regtable([reg0, reg1, reg2, reg3, reg4], stat='tstat', sig='coeff')
tbl.render()

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/finance_byu/regtables.py:176: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  out = out.append(obs)
/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/finance_byu/regtables.py:178: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  out = out.append(rsqs)


er0       er1       er2       er3       er4
                                                            
Intercept  -0.857***  -0.216**     0.107  0.327***  0.595***
             (-6.49)   (-2.57)    (1.63)    (5.36)    (6.33)
exmkt       1.488***  1.215***  1.096***  1.030***  1.092***
             (60.87)   (78.18)   (89.76)   (91.05)   (62.70)
Obs             1149      1149      1149      1149      1149
Rsq             0.76      0.84      0.88      0.88      0.77

In [ ]:
from finance_byu.statistics import GRS

grsstat, pval, tbl = GRS(port, names, ['exmkt'])

print(f'GRS = {grsstat:.2f} and p-value = {pval:.5g}\n')

### 3.
Interpret the regression results from question 2). What can you infer? Can you reject that the CAPM holds? Is the market portfolio the tangency portfolio?

The coefficient on 'exmkt' gives us an estimate of the beta of each portfolio. The numbers we really want to look at is the intercept and it's t-statistic. This intercept represents alpha. According to the CAPM, alpha should be zero. We see that alpha is more positive for higher momentum portfolios and more negative for portfolios with less momentum. The associated t-statistics get more extreme (and statistically significant) as the portfolios get more extreme (meaning more or less momentum). This means that momentum is a pretty good predicter of alpha, which gives us pretty strong evidence that we should reject the CAPM. This means the market portfolio is not the targency portfolio and that the tangency portfolio should hold more momentum stocks.

### 4.
Create a spread portfolio:
$$
r_{spread,t} = r_{4t} - r_{0t}.
$$
Test the CAPM using this portfolio. Can you reject the CAPM?

In [25]:
port['spread'] = port['p4'] - port['p0']
reg_s = smf.ols('spread ~ exmkt', data=port).fit()
reg_s.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 spread   R-squared:                       0.133
Model:                            OLS   Adj. R-squared:                  0.132
Method:                 Least Squares   F-statistic:                     175.5
Date:                Thu, 09 Feb 2023   Prob (F-statistic):           2.25e-37
Time:                        12:37:08   Log-Likelihood:                -3574.5
No. Observations:                1149   AIC:                             7153.
Df Residuals:                    1147   BIC:                             7163.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.4521      0.162      8.988      0.000       1.135       1.769
exmkt         -0.3963      0.030    -13.247      0.000      -0.455      -0.338
==============================================================================
Omnibus:                      617.941   Durbin-Watson:                   1.948
Prob(Omnibus):                  0.000   Jarque-Bera (JB):            13940.365
Skew:                          -1.991   Prob(JB):                         0.00
Kurtosis:                      19.593   Cond. No.                         5.44
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

This regression gives us a high intercept (alpha) with a very high t-statistic (almost 9). It would be hard not to reject the CAPM looking at these results. They show a clear to get returns that are not explained by market beta.

### 5.
Estimate the security market line using the data available for this homework. Specifically, estimate the following line:
<br>
$$
E(r_p) = r_f + \beta_{p}\bigl[E(r_M) - r_f\bigr]
$$
<br>
You don't need to plot the estimated line, but report your estimates of $r_f$ and $E(r_M) - r_f$.

In [57]:
beta = tbl.get_coefficients().loc['exmkt'].values
ex_ret = summary(port[['p0', 'p1', 'p2', 'p3', 'p4']]).loc['mean'].values
temp_df = pd.DataFrame({'beta':beta, 'ex_ret':ex_ret})
temp_df

beta    ex_ret
0  1.487981  0.392829
1  1.215244  0.853671
2  1.096231  1.098152
3  1.029521  1.273673
4  1.091710  1.583043

In [59]:
reg = smf.ols('ex_ret ~ beta', data=temp_df).fit()
reg.summary()

/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/statsmodels/stats/stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 ex_ret   R-squared:                       0.802
Model:                            OLS   Adj. R-squared:                  0.736
Method:                 Least Squares   F-statistic:                     12.18
Date:                Thu, 09 Feb 2023   Prob (F-statistic):             0.0398
Time:                        13:04:56   Log-Likelihood:                 1.5201
No. Observations:                   5   AIC:                            0.9599
Df Residuals:                       3   BIC:                            0.1787
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      3.6475      0.754      4.837      0.017       1.247       6.048
beta          -2.2018      0.631     -3.490      0.040      -4.210      -0.194
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   1.379
Prob(Omnibus):                    nan   Jarque-Bera (JB):                1.182
Skew:                           1.184   Prob(JB):                        0.554
Kurtosis:                       2.738   Cond. No.                         14.8
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

My estimate of $r_f$ is 3.6475 and my estimate for $E(r_M) - r_f$ is -2.2018.

### 6.
Why is the intercept in time series CAPM regression called an *average abnormal return*? Briefly explain.

It is the average return that is not explained by taking on risk. The more risk you take on, the more expected return, but even taking on no risk gives a small return. The intercept is the return not explained by beta.